std/os/index.zig

```diff
@@ -2512,6 +2512,49 @@ pub fn posixGetSockOptConnectError(sockfd: i32) PosixConnectError!void {
     }
 }
 
+pub const PosixFcntlError = error{
+    /// EACCES: Operation is prohibited by locks held by other processes.
+    PermissionDenied,
+
+    /// EAGAIN: The operation is prohibited because the file has been memory-
+    /// mapped by another process.
+    SystemResources,
+
+    /// EBUSY: cmd is F_SETPIPE_SZ and the new pipe capacity specified in arg
+    /// is smaller than the amount of buffer space currently used to
+    /// store data in the pipe.
+    /// OR cmd is F_ADD_SEALS, arg includes F_SEAL_WRITE, and there
+    /// exists a writable, shared mapping on the file referred to by fd.
+    FileBusy,
+
+    Unexpected,
+};
+
+pub fn posixFcntl(fd: i32, cmd: i32, a: ...) PosixFcntlError!i32 {
```

Comment:
For fcntl, it might make sense to not provide a `posixFcntl` wrapper since the errors are different for every subcommand. Might be better to leave the generic fcntl as `posix.fcntl(...)`. However it would make sense to provide `os.posixFcntlSetPipeSz`, `os.posixFcntlAddSeals`, etc. These would accept a finite set of usefully typed arguments, and hard-code the cmd parameter, and each one would have a customized error set applicable to only that cmd. This would probably be a good time to start file `std/os/fcntl.zig`.

Reply:
Even though there are different errors for each sub command, `fcntl` is too OS and domain specific to create a specific error for each OS level error. For POSIX errors, I think that is fine, but for each OS, the caller of `fcntl` should know what they are getting into.

Likewise, I am creating specific functions in `net.zig` that focus on using `fcntl` to set sockets non blocking, etc -- those functions will support all of the small issues with each OS.

Reply:
In this case I don't think we should have `PosixFcntlError`. It's misleading since the error set does not map correctly to the function call. The posix-specific code in net.zig can call `os.posix.fcntl2` directly and deal with the errno manually.

Reply:
okay, I agree with this. I will remove `PosixFcntlError` and `posixFcntl`.

Reply:
The philosophy here is: don't make leaky abstractions. Better to leave something unabstracted than to abstract it in a problematic way.

Reply:
I agree -- I was just trying to follow in your footsteps with the other calls.
`fcntl` is definitely one of those syscalls that is wide in scope. Glad to get the distinction.